plutus-playground-client/src/Chain/BlockchainExploration.purs

```diff
@@ -93,13 +93,19 @@ blockchainExploration blockchain =
 
     columnHeading FeeIx = "Fee"
     columnHeading ForgeIx = "Forge"
-    columnHeading (OwnerIx owner hash) = "Wallet #" <> show owner
-    columnHeading (ScriptIx owner hash) = "Script #" <> owner
+    columnHeading (OwnerIx owner hash) = "Wallet #" <> abbreviate (show owner)
+    columnHeading (ScriptIx owner hash) = "Script #" <> abbreviate owner
 
     columnSubheading FeeIx = ""
     columnSubheading ForgeIx = ""
-    columnSubheading (OwnerIx owner hash) = "Tx/" <> String.take 10 hash <> "..."
-    columnSubheading (ScriptIx owner hash) = "Tx/" <> String.take 10 hash <> "..."
+    columnSubheading (OwnerIx owner hash) = "Tx/" <> abbreviate hash
+    columnSubheading (ScriptIx owner hash) = "Tx/" <> abbreviate hash
+
+    abbreviate :: String -> String
+    abbreviate str =
+      if String.length str <= 7
+      then str
+      else String.take 10 str <> "..."
```

Comment:
why is this take 10? Surely a string of length 8 will have the entire string and the three dots?

Reply:
Doh! I was rushing yesterday, clearly. Fixed, with added quickcheck tests.